public/app/core/utils/file_export.ts

```diff
@@ -93,14 +112,24 @@ export function convertSeriesListToCsvColumns(seriesList, dateTimeFormat = DEFAU
   // make text
   for (let i = 0; i < seriesList[0].datapoints.length; i += 1) {
     const timestamp = moment(seriesList[0].datapoints[i][POINT_TIME_INDEX]).format(dateTimeFormat);
-    text += formatRow(
-      [timestamp].concat(
-        seriesList.map(series => {
-          return series.datapoints[i][POINT_VALUE_INDEX];
-        })
-      ),
-      i < seriesList[0].datapoints.length - 1
-    );
+
+    // if a value is null and removeNullValues is true we don't add the text
+    let hasNullValues = false;
+
+    seriesList.map(series => {
+      hasNullValues = series.datapoints[i][POINT_VALUE_INDEX] ? false : true;
+    });
+
+    if ((!hasNullValues && removeNullValues) || !removeNullValues) {
```

Comment:
can you explain what you are trying to do here??? maybe add a few unit tests. Not sure what your trying to do with this code. if any value is null remove the column?